lib/index.js

```diff
@@ -79,24 +84,34 @@ internals.implementation = function (server, options) {
             request.plugins[pkg.name] = { extraInfo: extraInfo };
           }
           var verifyOptions = options.verifyOptions || {};
-          JWT.verify(token, key, verifyOptions, function (err, decoded) {
-            if (err) {
-              return reply(raiseError('unauthorized', 'Invalid token', tokenType), null, { credentials: null });
-            }
-            else { // see: http://hapijs.com/tutorials/auth for validateFunc signature
-              options.validateFunc(decoded, request, function (err, valid, credentials) { // bring your own checks
-                if (err) {
-                  return reply(raiseError('wrap', err));
-                }
-                else if (!valid) {
-                  return reply(raiseError('unauthorized', 'Invalid credentials', tokenType), null, { credentials: credentials || decoded });
-                }
-                else {
-                  return reply.continue({ credentials: credentials || decoded, artifacts: token });
+          var keys = (internals.isArray(key)) ? key : [ key ];
+          var keysTried = 0;
+
+          keys.some(function (key) {
+            JWT.verify(token, key, verifyOptions, function (err, decoded) {
+              if (err) {
+                keysTried++;
+                if (keysTried >= keys.length) {
```

Comment:
@gangstead I don't see a situation where `keysTried` _would_ be greater than `keys.length` ... ? 🤔